include/rotatingMHD/solver_class.h

```diff
@@ -344,19 +432,45 @@ LinearAlgebra::MPI::Vector        projection_step_rhs;
  */
 double                            norm_projection_step_rhs;
 
+/*!
+ * @brief The system matrix of the zeroth step.
+ *
+ * @details The system matrix corresponds to the stiffness matrix of the
+ * Lagrange multiplier field. Its entries only change if the
+ * spatial discretization changes. It is stored in memory because
+ * otherwise an assembly would be required if the timestep changes.
+ *
+ */
+LinearAlgebra::MPI::SparseMatrix  zeroth_step_system_matrix;
+
+/*!
+ * @brief Vector representing the right-hand side of the linear system
+ * of the zeroth step performed to compute admissible initial conditions
+ * for the Lagrange multiplier.
+ *
+ */
+LinearAlgebra::MPI::Vector        zeroth_step_rhs;
```

Comment:
What is the purpose of these variables? They are only used once...

Reply:
I find it more convenient to have them as members in order to avoid duplicate methods. If they were local variables declared inside `initialization_step` they'd have to be passed to their own isolated setup and assembly methods, which would lead to duplicated code. If memory is your concern, one could call their `clear()` methods at the end of the initialization step.

Reply:
Yes a call of `clear()` would be good to free memory.